doc/api/async_hooks.md

```diff
@@ -65,6 +66,11 @@ function after(asyncId) { }
 
 // destroy is called when an AsyncWrap instance is destroyed.
 function destroy(asyncId) { }
+
+// promiseResolve is called only for promise resources, when the
+// `resolve` function passed to the `Promise` constructor is invoked
+// (either directly or through other means of resolving a promise).
+function promiseResolve(asyncId) { }
```

Comment:
Would it be called *before* or *after* the `resolve` function is invoked? The distinction may be important for measuring timing.

Reply:
@jasnell I’m pretty sure that’s actually irrelevant, since the `resolve` method doesn’t actually do any synchronously visible work.

Reply:
This should also likely include some clarification about the relationship (if any) this has to the `before` and `after` emits. An example that shows what kind of ordering of these events to expect when doing a `new Promise((resolve) => resolve(true)).then((a) => {})` would be worthwhile.

Reply:
@addaleax 
> I’m pretty sure that’s actually irrelevant, since the resolve method doesn’t actually do any synchronously visible work.

Then I would be explicit about that in the docs.

Reply:
I’ve added a note about that and added an example